xarray/core/netcdftimeindex.py

```diff
@@ -0,0 +1,180 @@
+import re
+from datetime import timedelta
+
+import numpy as np
+import pandas as pd
+
+from pandas.lib import isscalar
+
+
+def named(name, pattern):
+    return '(?P<' + name + '>' + pattern + ')'
+
+
+def optional(x):
+    return '(?:' + x + ')?'
+
+
+def trailing_optional(xs):
+    if not xs:
+        return ''
+    return xs[0] + optional(trailing_optional(xs[1:]))
+
+
+def build_pattern(date_sep='\-', datetime_sep='T', time_sep='\:'):
+    pieces = [(None, 'year', '\d{4}'),
```

Comment:
Then let's not bother until someone asks. Per [Wikipedia's ISO 8601](https://en.wikipedia.org/wiki/ISO_8601#Years) you can optionally use an expanded year representation with `+` and `-`. I don't think they would always be necessary but I haven't read the original document (which unfortunately I think is not available only).